### ISSUE TITLE
a11y: improve input group accessibility

### ISSUE THREAD

**Issue by desig9stein** [IgniteUI/igniteui-angular#8925]
## Description  

Provide an accessible experience for igx-input-group.

1. If the input-group has a label, it should automatically be used as the `aria-label` for the input filed
2. if the input-group has `hint` it should be added to the `aria-describedby` list of the input field
3. `aria-invalid` should be added and automatically updated based on the input's validity state.
4. `aria-required` should be added if the input is a required field. 
5. To notify the user that there is a problem, Add `role="alert"` to the error message.

NOTE:  According to [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/forms/alerts) `aria-required` is required only for browsers that do not yet support ARIA.

 * igniteui-angular version: 11.1.x
 * browser: ALL

## Attachments  
Attach a sample if available, and screenshots, if applicable.

**Comment by Lipata** [comment]
Moved to samples repo: https://github.com/IgniteUI/igniteui-angular-samples/issues/2739
